src/Stateless/Reflection/MethodInfo.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Stateless.Reflection
+{
+    /// <summary>
+    /// Describes an internal method through the reflection API.
+    /// </summary>
+    public class MethodInfo
```

Comment:
Argh.  Yeah, that's not good to overlap with the system class.  InvocationInfo is doable, but thinking outside the box for a second - can we get rid of the MethodInfo class completely and just add the "calculate the description" method to MethodDescription, and use it for both the main state machine and the reflection API?